tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -0,0 +1,771 @@
+	"github.com/openshift-kni/eco-gotests/tests/cnf/core/network/metallb/internal/tsparams"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFailure, func() {
+	var (
+		externalAdvertisedIPv4Routes = []string{"192.168.100.0/24", "192.168.200.0/24"}
+		externalAdvertisedIPv6Routes = []string{"2001:100::0/64", "2001:200::0/64"}
+		hubIPaddresses               = []string{"172.16.0.10", "172.16.0.11"}
+		err                          error
+	)
+
+	BeforeAll(func() {
+
+		By("Getting MetalLb load balancer ip addresses")
+		ipv4metalLbIPList, ipv6metalLbIPList, err = metallbenv.GetMetalLbIPByIPStack()
+		Expect(err).ToNot(HaveOccurred(), tsparams.MlbAddressListError)
+
+		By("List CNF worker nodes in cluster")
+		cnfWorkerNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Failed to discover worker nodes")
+
+		By("Selecting worker node for BGP tests")
+		workerLabelMap, workerNodeList = setWorkerNodeListAndLabelForBfdTests(cnfWorkerNodeList, metalLbTestsLabel)
+		ipv4NodeAddrList, err = nodes.ListExternalIPv4Networks(
+			APIClient, metav1.ListOptions{LabelSelector: labels.Set(workerLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Failed to collect external nodes ip addresses")
+
+		err = metallbenv.IsEnvVarMetalLbIPinNodeExtNetRange(ipv4NodeAddrList, ipv4metalLbIPList, nil)
+		Expect(err).ToNot(HaveOccurred(), "Failed to validate metalLb exported ip address")
+
+		By("Listing master nodes")
+		masterNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.ControlPlaneLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Fail to list master nodes")
+		Expect(len(masterNodeList)).To(BeNumerically(">", 0),
+			"Failed to detect master nodes")
+	})
+
+	Context("IBGP Single hop", func() {
+
+		var (
+			nodeAddrList []string
+			addressPool  []string
+			err          error
+			frrk8sPods   []*pod.Builder
+		)
+
+		BeforeAll(func() {
+			By("Setting test iteration parameters")
+			_, _, _, nodeAddrList, addressPool, _, err =
+				metallbenv.DefineIterationParams(
+					ipv4metalLbIPList, ipv6metalLbIPList, ipv4NodeAddrList, ipv6NodeAddrList, "IPv4")
+			Expect(err).ToNot(HaveOccurred(), "Fail to set iteration parameters")
+
+		})
+
+		AfterEach(func() {
+			By("Clean metallb operator and test namespaces")
+			resetOperatorAndTestNS()
+		})
+
+		It("Verify that prefixes configured with alwaysBlock are not received by the FRR speakers",
+			reportxml.ID("74270"), func() {
+				By("Creating a new instance of MetalLB Speakers on workers blocking specific incoming prefixes")
+				err := createNewMetalLbDaemonSetAndWaitUntilItsRunningWithAlwaysBlock(tsparams.DefaultTimeout,
+					workerLabelMap, []string{externalAdvertisedIPv4Routes[0]})
```

Comment:
nit: declare `prefixToBlock := externalAdvertisedIPv4Routes[0]` and then use it here and elsewhere, will make it more readable

Reply:
Added